src/adapter/src/catalog/state.rs

```diff
@@ -1419,14 +1419,12 @@ impl CatalogState {
                 cluster_id: in_cluster,
             }),
             Plan::CreateType(CreateTypePlan { typ, .. }) => {
-                let desc = match typ.inner.desc(&session_catalog) {
-                    Ok(desc) => desc,
-                    Err(err) => return Err((err.into(), cached_expr)),
-                };
+                if let Err(err) = typ.inner.desc(&session_catalog) {
+                    return Err((err.into(), cached_expr));
+                }
```

Comment:
Is this check important? If it is, we should have a comment explaining why. Otherwise, we should remove it.